### PR TITLE
tiflash: Add desc of new TiFlash storage settings and guidelines for upgrading in v4.0.9 (#4290)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #4290 to release-4.0

---

Signed-off-by: JaySon-Huang <tshent@qq.com>

### What is changed, added or deleted? (Required)

In v4.0.9, TiFlash supports new configurations for better performance when it is deployed on multi-disk node.

This PR
* Add some description about the new configurations.
* Add guidelines for how to enable those new configurations after upgrading the TiDB cluster.

Please do NOT merge this PR until v4.0.9 is ready for release~

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4965
- Other reference link(s): https://github.com/pingcap/tiup/pull/931

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Review by TomShawn — APPROVED** [review]

**Comment by TomShawn** [comment]
@JaySon-Huang PTAL whether I have correctly applied v4.0-specific changes, thanks!

File: tiflash/tiflash-configuration.md [hunk 1]
```diff
@@ -35,25 +35,70 @@ This section introduces the configuration parameters of TiFlash.
...
+
+## The path in which the TiFlash temporary files are stored. By default it is the first directory in `path`
+## or in `storage.latest.dir` appended with "/tmp".
+# tmp_path = "/tidb-data/tiflash-9000/tmp"
+
+## Storage paths settings take effect starting from v4.0.9
+[storage]
+    ## Limits the total write rate of background tasks in bytes per second. 0 means no limit.
+    bg_task_io_rate_limit = 0
+
```
Comment (JaySon-Huang) [inline tiflash/tiflash-configuration.md:70]:
```suggestion
```

**Review by JaySon-Huang — COMMENTED** [review]
Remove the configuration not introduced to v4.0

File: tiflash/tiflash-configuration.md [hunk 2]
```diff
@@ -35,25 +35,67 @@ This section introduces the configuration parameters of TiFlash.
...
+## The TiFlash TCP service port.
+tcp_port = 9000
+## The TiFlash HTTP service port.
+http_port = 8123
+## The cache size limit of the metadata of a data block. Generally, you do not need to change this value.
+mark_cache_size = 5368709120
+## The cache size limit of the min-max index of a data block. Generally, you do not need to change this value.
+minmax_index_cache_size = 5368709120
+## The cache size limit of the DeltaIndex. The default value is 0, which means no limit.
+delta_index_cache_size = 0
```
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:49]:
```suggestion
```

**Pull request merged**